catalyst-gateway/bin/src/db/index/queries/staked_ada/get_txo_by_stake_address.rs

```diff
@@ -69,12 +82,64 @@ impl GetTxoByStakeAddressQuery {
     /// Executes a get txo by stake address query.
     pub(crate) async fn execute(
         session: &CassandraSession, params: GetTxoByStakeAddressQueryParams,
-    ) -> anyhow::Result<TypedRowStream<GetTxoByStakeAddressQuery>> {
-        let iter = session
-            .execute_iter(PreparedSelectQuery::TxoByStakeAddress, params)
+    ) -> anyhow::Result<Vec<GetTxoByStakeAddressQuery>> {
+        if session.is_persistent() {
```

Comment:
Its an anti pattern to make the queries themselves session aware.
Any operation thats dependent on the session should be part of the session.
For example, this should just be:

```
            if let Some(rows) = session.caches.txo_by_stake.get(&params.stake_address) {
                return Ok(rows);
            }
```